contracts/upgradeable_contracts/modules/interest/AAVEInterestERC20.sol

```diff
@@ -0,0 +1,240 @@
+pragma solidity 0.7.5;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "../../../interfaces/IAToken.sol";
+import "../../../interfaces/IOwnable.sol";
+import "../../../interfaces/IInterestReceiver.sol";
+import "../../../interfaces/IInterestImplementation.sol";
+import "../../../interfaces/ILendingPool.sol";
+import "../MediatorOwnableModule.sol";
+
+/**
+ * @title AAVEInterestERC20
+ * @dev This contract contains token-specific logic for investing ERC20 tokens into AAVE protocol.
+ */
+contract AAVEInterestERC20 is IInterestImplementation, MediatorOwnableModule {
+    using SafeMath for uint256;
+
+    event PaidInterest(address indexed token, address to, uint256 value);
+
+    struct InterestParams {
+        IAToken aToken;
+        uint96 dust;
+        uint256 investedAmount;
+        address interestReceiver;
+        uint256 minInterestPaid;
+    }
+
+    mapping(address => InterestParams) public interestParams;
+
+    constructor(address _omnibridge, address _owner) MediatorOwnableModule(_omnibridge, _owner) {}
+
+    /**
+     * @dev Tells the module interface version that this contract supports.
+     * @return major value of the version
+     * @return minor value of the version
+     * @return patch value of the version
+     */
+    function getModuleInterfacesVersion()
+        external
+        pure
+        returns (
+            uint64 major,
+            uint64 minor,
+            uint64 patch
+        )
+    {
+        return (1, 0, 0);
+    }
+
+    /**
+     * @dev Tells the address of the LendingPool contract in the Ethereum Mainnet.
+     */
+    function lendingPool() public pure virtual returns (ILendingPool) {
+        return ILendingPool(0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9);
+    }
+
+    /**
+     * @dev Enables support for interest earning through specific aToken.
+     * @param _aToken address of the aToken contract. Underlying token address is derived from this contract.
+     * @param _dust small amount of underlying tokens that cannot be paid as an interest. Accounts for possible truncation errors.
+     * @param _interestReceiver address of the interest receiver for underlying token and associated COMP tokens.
+     * @param _minInterestPaid min amount of underlying tokens to be paid as an interest.
+     */
+    function enableInterestToken(
+        IAToken _aToken,
+        uint96 _dust,
+        address _interestReceiver,
+        uint256 _minInterestPaid
+    ) external onlyOwner {
+        address token = _aToken.UNDERLYING_ASSET_ADDRESS();
```

Comment:
Will it be easier to specify in the method `enableInterestToken` the locked asset address instead of the corresponding AAVE token? The method could find an aToken by using the lending pool's method `getReserveData`. For example, for USDC token `0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48` it returns the following:

![image](https://user-images.githubusercontent.com/20793260/123541611-11bdbb80-d74e-11eb-9288-649f86eb4953.png)

Reply:
I think it would be a little bit simpler to use, but calling `getReserveData` is possible only with enabled abi coder v2. I think it would be better to not enable it for now, to avoid possible verification difficulties.

Reply:
But we can get the data without the abi coder v2. You can use `call` to invoke the method through its signature and get the data as a blob.    After that the aToken address can be received from this blob by some shift. Is it possible?

Reply:
Yeah, I have found a tricky way of doing this without directly writing it in inline assembly.